### ISSUE TITLE
Early stopping?

### ISSUE THREAD

**Issue by Jesse-Islam** [autonlab/DeepSurvivalMachines#47]
Hi,

I was wondering if there's an early stopping parameter for dsm that's already implemented, or an ideal work around that you would use now to get it going. I've been digging through the source code and can't seem to find any. I imagine its not too complicated as you're extending pytorch, but I'm also not sure what the best way to make it extend your functions would be.

**Comment by chiragnagpal** [comment]
@Jesse-Islam The code does implement a somewhat memory inefficient version of early stopping, check lines 187-207 https://github.com/autonlab/DeepSurvivalMachines/blob/2801c6c63a48195821733245c85f9a90bd21d446/dsm/utilities.py#L187https://github.com/autonlab/DeepSurvivalMachines/blob/2801c6c63a48195821733245c85f9a90bd21d446/dsm/utilities.py#L187
@Jeanselme is working on a more efficient version here: #37

**Comment by Jesse-Islam** [comment]
Awesome! So I could add a minimum delta loss required and increase patience from 2 to whatever i'd like... The only issue I see with the current implementation is that (If I'm following correctly) is that the currently implementation does not slide the window (i.e. 1-10, then 11-20, then 21-30)

Thanks for the help!

**Comment by chiragnagpal** [comment]
sure, anytime :) you can also pass a tuple of validation data to `val_data` to compute the loss on a heldout validation set

**Comment by Jesse-Islam** [comment]
Would you suggest that over using the vsize parameter? for example setting it to .20 so that the validation would be 20% of the data fed in to train.

**Comment by chiragnagpal** [comment]
either is ok -- visze would randomly sample the val set from the training data.. if you want something that is reproducible across runs, i'd recommend just pass the val_data explicitly
